### ISSUE TITLE
Feature: Permissive options passthrough

### ISSUE THREAD

**Issue by ainaadekunle** [aai-research-lab/FastMDAnalysis#6]
# Feature: Permissive options passthrough + alias mapping + hooks

## Summary
Implement a **permissive options passthrough** so users can supply MDTraj/scikit-learn–style options across **all analyses**.  
fastmda should **map aliases**, apply **pre/post hooks** where upstream lacks direct kwargs (e.g., `rmsd.align`, `rmsf.per_residue`), and provide a **`--strict`** mode for schema validation.

## Scope (all analyses)
`rmsd`, `rmsf`, `rg`, `hbonds`, `ss`, `sasa`, `dimred`, `cluster`.

## Motivation
- Users expect MDTraj/scikit-learn kwargs to “just work.”
- Current behavior drops unknown keys even when a sensible mapping exists.
- Improves **API/CLI parity**, **usability**, and **reproducibility** without teaching a new options vocabulary.

## Proposed solution
- **Forwarding utility**
  - Introspect target signatures; forward accepted kwargs.
  - **Alias mapping** (e.g., `ref → reference`, `atoms → atom_indices`).
  - **pre_hook** (e.g., `align: true` → `traj.superpose(...)`), **post_hook** (e.g., `per_residue: true` → residue aggregation).
  - Unknown keys: **log** (non-strict) or **raise** (`--strict`).

- **Module specifics**
  - **rmsd:** aliases `ref`, `atoms`; pre-align if `align: true` (default).
  - **rmsf:** aliases `atoms`, `reference`; post-aggregate if `per_residue: true`.
  - **rg:** alias `atoms`; forward `mass_weighted` (if supported).
  - **hbonds:** map `distance_cutoff_nm`, `angle_cutoff_deg`, `sidechain_only`, `exclude_water`, `periodic`; hook to (re)create bonds if topology lacks them.
  - **ss:** `algorithm: dssp`, optional `mkdssp_path`; expose per-residue/per-frame outputs.
  - **sasa:** map `probe_radius_nm`, `n_sphere_points`, `atoms/selection`.
  - **dimred:** `method ∈ {pca, mds, tsne}`; forward `n_components`, `random_state`, `perplexity`, `n_iter`, `metric/dissimilarity`.
  - **cluster:** `method ∈ {dbscan, kmeans, hierarchical}`; forward sklearn kwargs (`n_clusters`, `linkage`, `random_state`, `n_init`, `eps`, `min_samples`).

- **Strictness**
  - Add **`--strict`** (CLI) and an API flag to toggle permissive vs. validating behavior.

## YAML example
```yaml
rmsd:
  ref: 0
  align: true
  atoms: "protein and name CA"

rmsf:
  per_residue: true
  atoms: "protein and name CA"

rg:
  atoms: "protein"
  mass_weighted: false

hbonds:
  distance_cutoff_nm: 0.25
  angle_cutoff_deg: 120
  sidechain_only: false
  exclude_water: true
  periodic: true

ss:
  algorithm: dssp
  mkdssp_path: "<optional>"

sasa:
  probe_radius_nm: 0.14
  n_sphere_points: 960
  atoms: "protein"

dimred:
  method: tsne
  n_components: 2
  perplexity: 35
  n_iter: 1500
  random_state: 42
  metric: euclidean   # for MDS: use dissimilarity: precomputed

cluster:
  method: kmeans      # or: hierarchical | dbscan
  n_clusters: 6
  random_state: 42
  linkage: ward       # hierarchical only
  n_init: "auto"      # kmeans only
  eps: 0.4            # dbscan only
  min_samples: 5      # dbscan only

**Comment by Paradoxicaly** [comment]
Created permissive options passthrough feature. Review newest pull request
